fdbrpc/SimExternalConnection.h

```diff
@@ -48,8 +48,19 @@ class SimExternalConnection final : public IConnection, public ReferenceCounted<
 	int write(SendBuffer const* buffer, int limit) override;
 	NetworkAddress getPeerAddress() const override;
 	UID getDebugID() const override;
+	static void addMockTCPEndpoint(const std::string& host,
```

Comment:
It looks like all of these added static methods have nothing to do with the rest of the class. It is only used to manipulate `hostnameToAddresses`, which is intended to be a global map. I think these should live outside of this class. `resolveTCPEndpoint ` can access that global map to find host->address mapping.

Reply:
Same reason. These classes should be simulation only.